torch_em/shallow2deep/prepare_shallow2deep.py

```diff
@@ -504,11 +509,94 @@ def random_points(
     return features[samples], labels[samples]
 
 
+def worst_tiles(
+    features, labels, rf_id,
+    forests, forests_per_stage,
+    sample_fraction_per_stage,
+    sampling_kwargs,
+    accumulate_samples=True,
+):
+    # check inputs
+    img_shape = sampling_kwargs.get('img_shape', None)
+    assert len(img_shape) in [2, 3], img_shape
+    tiles_shape = sampling_kwargs.get('tiles_shape', None)
+    assert len(tiles_shape) in [2, 3], tiles_shape
+
+    # get the corresponding random forest from the last stage
+    # and predict with it
+    last_forest = forests[rf_id - forests_per_stage]
+    pred = last_forest.predict_proba(features)
+
+    # labels to one-hot encoding
+    unique, inverse = np.unique(labels, return_inverse=True)
+    onehot = np.eye(unique.shape[0])[inverse]
+
+    # compute the difference between labels and prediction
+    diff = np.abs(onehot - pred).sum(axis=1)
+    assert len(diff) == len(features)
+
+    # reshape diff to image shape and apply convolution with 1-kernel
+    diff_img = torch.Tensor(diff.reshape(img_shape)[None, None])
+    kernel = torch.Tensor(np.ones(tiles_shape)[None, None])
+    diff_img_smooth = torch.nn.functional.conv2d(diff_img, weight=kernel) if len(img_shape)==2 \
```

Comment:
I think using torch for the convolution here is a bit overkill. Two points:
- Do we actually want to apply a uniform kernel here, or do we rather want to apply smoothing with a large sigma (for this can use `vigra.filters.gaussianSmoothing`)
- If we do want the uniform kernel you can use the convolve functionality from scipy, speed doesn't matter here and it's easier to read because you don't need to convert to torch tensors: `scipy.ndimage.convolve`.

Reply:
Sure, I'll replace torch. About the question what's better, I don't know to be honest. What do you think? Maybe it's possible to allow the smoothing function to be given as an input. Or do you think that's a bit overkill?

Reply:
> What do you think? Maybe it's possible to allow the smoothing function to be given as an input. Or do you think that's a bit overkill?

Maybe we can keep it a bit simpler and have a single parameter, that is either a scalar (then we interpret it as sigma and use gaussian smoothing) or a tuple (then we interpret it as tile shape and apply the uniform kernel)